frontend/src/model/schema/methodObjects/FdScore.ts

```diff
@@ -60,20 +60,36 @@ export default class FdScore {
    * @returns a ranking score used different ranking approaches 
    */
   private calculate(): number {
+    let lengthScore: number = (window as any).DEFAULT_RANKING_WEIGHTS.length != 0 
```

Comment:
Sind ein wenig viele ähnliche Zeilen, würde sowas vorschlagen:

```typescript
    const scoreTypes = {
      length: this.fdLengthScore,
      keyValue: this.fdKeyValueScore,
      position: this.fdPositionScore,
      redundanceTeam: this.fdRedundanceScoreTeam,
      redundanceWeiLink: this.fdRedundanceScoreWeiLink,
      redundanceMetanome: this.fdRedundanceScoreMetanome,
      similarity: this.fdSimilarityScore
    };
    let score = 0
    for (const [weightName, scoreFunction] of Object.entries(scoreTypes)) {
      const weight: number = (window as any)[weightName] ?? 0
      if (weight !== 0) score += weight * scoreFunction()
    }
    return score
```

Reply:
Musste es ein bisschen anders machen, weil man noch this an die Funktion geben musste. Aber ansonsten ist es umgesetzt